### ISSUE TITLE
make fails on button.o when compiling Out-Of-Tree, file shaders.gen.h not found

### ISSUE THREAD

**Issue by Mutos** [naev/naev#1147]
`make` fails with the following error :

```
Making all in widget
  CC    button.o
In file included from ../../../../../src/font.h:11:0,
                 from ../../../../../src/tk/widget/text.h:11,
                 from ../../../../../src/tk/widget.h:12,
                 from ../../../../../src/tk/toolkit_priv.h:14,
                 from ../../../../../src/tk/widget/button.c:12:
../../../../../src/opengl.h:30:10: fatal error: shaders.gen.h: Aucun fichier ou dossier de ce type
 #include "shaders.gen.h"
          ^~~~~~~~~~~~~~~
compilation terminated.
Makefile:456: recipe for target 'button.o' failed
make[4]: *** [button.o] Error 1
Makefile:426: recipe for target 'all-recursive' failed
make[3]: *** [all-recursive] Error 1
Makefile:1137: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
Makefile:837: recipe for target 'all' failed
make[1]: *** [all] Error 2
Makefile:667: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

```

**Comment by Mutos** [comment]
Removing the line to see if it's actually used gives the following error :

```
Making all in widget
  CC    button.o
  CC    checkbox.o
  CC    cust.o
  CC    fader.o
  CC    image.o
  CC    imagearray.o
  CC    input.o
  CC    list.o
  CC    rect.o
  CC    tabwin.o
  CC    text.o
ar: le modificateur « u » est ignoré puisque « D » est le défaut (consultez « U »)
  CC    ai.o
  CC    array.o
  CC    background.o
../../../src/background.c: In function ‘background_renderStars’:
../../../src/background.c:167:17: error: ‘shaders’ undeclared (first use in this function); did you mean ‘stderr’?
    glUseProgram(shaders.stars.program);
                 ^~~~~~~
                 stderr
../../../src/background.c:167:17: note: each undeclared identifier is reported only once for each function it appears in
Makefile:1099: recipe for target 'background.o' failed
make[3]: *** [background.o] Error 1
Makefile:1137: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
Makefile:837: recipe for target 'all' failed
make[1]: *** [all] Error 2
Makefile:667: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

```

**Comment by Mutos** [comment]
An internet search gives me nothing out of NAEV, but the following commit : 4cc1f4f53dba9688945110b7707d830049356a6f or "Make sure shaders.gen.h is generated before any files that depend on it". It has been merged into `master` branch by February, 2020, so it should be OK...

**Comment by Mutos** [comment]
Seems there is a `src/shaders_c_gen.c` file that must be compiled into an executable that generates `src/shaders.gen.h` and `src/shaders.gen.c`...

**Comment by Mutos** [comment]
Tried on another machine, same Ubuntu, same error, still no idea of how I could solve that...

**Comment by Mutos** [comment]
Does anyone else encounter this issue ?

**Comment by Mutos** [comment]
A wild guess : would it stem from the OOT compile ?

Now, I need the OOT compile to be able to develop SoC under Windows and Linux...

So the question becomes : what exactly is broken in OOT compile ?

**Comment by Mutos** [comment]
**_Very dirty workaround_**, but in no case a solution :
- Compile in NAEV's root folder;
- Copy `src/shaders_c_gen.c` and `src/shaders_c_gen.h` to any `bin/<architecture>` you need to compile on;
- Run `make distclean` from NAEV's root folder, this will remove the files you just copied from `src` folder;
- Perform whichever OOT compiles you need, they should now work...

I just tested it for `lin64`, having no other system at hand...

**Comment by Mutos** [comment]
Opposite to issue #1148, the idea could be to check `src` from compilation root folder instead of NAEV's root folder.

**Comment by bobbens** [comment]
It should be building the shaders.gen.h for you. What do you mean exactly by building out-of-tree? When using source tarballs, it should package shaders.gen.[ch] for you.

**Comment by Mutos** [comment]
Hi bobbens,

Building OOT is making a `bin/<architecture>` folder for each architecture you're compiling on.

Let's say in my case, I have `bin/lin64`, `bin/win32` and `bin/win64`. I run `autogen.sh` from the regular NAEV root, then I `cd` to the architecture folder I'm compiling on and I run `../../configure`, then `make` from that folder.

This way, I get a `bin` folder with 3 subfolders, each holding the NAEV executable for the right architecture, without different compiles messing up everything in their common folder...

**Comment by bobbens** [comment]
Should be fixed in 791daa768dcdb25a038f4fdd54db8b2663411914 , or at least I was able to on my machine compile like you do.

**Comment by Mutos** [comment]
Thanks bobbens, it works ! I'll look and see how you did it to learn something about the autogen/configure/make toolchain ^-^

**Comment by Mutos** [comment]
So I close the issue...
